controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -463,14 +492,9 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 			Bootstrap: clusterv1.Bootstrap{
 				ConfigRef: bootstrapRef,
 			},
+			FailureDomain: fd,
 		},
 	}
-	machine.Spec.FailureDomain = fd
-
-	owner := metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane"))
-	if owner != nil {
-		machine.SetOwnerReferences([]metav1.OwnerReference{*owner})
-	}
```

Comment:
Moved the setting of the owner ref into the definition of the machine rather than setting afterwards.